app/src/main/java/com/android/bookswap/ui/Addbook/listbookview.kt

```diff
@@ -0,0 +1,52 @@
+package com.android.bookswap.ui.Addbook
+
+import com.android.bookswap.data.DataBook
+import com.android.bookswap.model.Booksrepository
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+// A view class that interacts with the Books repository to manage a list of books.
+open class listToBooksView(private val repository: Booksrepository) {
+
+  // MutableStateFlow to hold the current list of books. It starts with an empty list.
+  private val books_ = MutableStateFlow<List<DataBook>>(emptyList())
+
+  // Publicly accessible read-only StateFlow of books, so that the UI can observe changes.
+  val books: StateFlow<List<DataBook>> = books_.asStateFlow()
+
+  // MutableStateFlow to hold the currently selected book. Initially, no book is selected (null).
+  private val selectbook_ = MutableStateFlow<DataBook?>(null)
+
+  // Publicly accessible read-only StateFlow of the selected book, so the UI can observe the
+  // selected book.
+  open val selectbook: StateFlow<DataBook?> = selectbook_.asStateFlow()
+
+  // Initialization block. Calls the repository's init() function when the class is instantiated.
+  init {
+    repository.init {}
+  }
+
+  // Function to generate a new unique identifier (UID) for a book, using the repository.
+  fun getNewUid(): String {
```

Comment:
We're currently using both Uid as string and UUID as identifier, this is a problem.
Everything should be UUID (firestore support this without a problem).
We will need to refactor this when we refactor firestore